qt/scientific_interfaces/Indirect/Elwin.cpp

```diff
@@ -50,6 +52,58 @@ int getNumberOfSpectra(std::string const &name) {
   return static_cast<int>(getADSMatrixWorkspace(name)->getNumberHistograms());
 }
 
+std::string extractLastOf(const std::string &str,
+                          const std::string &delimiter) {
+  const auto cutIndex = str.rfind(delimiter);
+  if (cutIndex != std::string::npos)
+    return str.substr(cutIndex + 1, str.size() - cutIndex);
+  return str;
+}
+
+template <typename Iterator, typename Predicate>
+std::vector<std::string> copyIf(Iterator const fromIterator,
+                                Iterator const toIterator,
+                                Predicate const &predicate) {
+  std::vector<std::string> newVector;
+  std::copy_if(fromIterator, toIterator, std::back_inserter(newVector),
+               predicate);
+  return newVector;
+}
+
+template <typename Iterator, typename Predicate>
+std::vector<std::string> transform(Iterator const fromIterator,
+                                   Iterator const toIterator,
+                                   Predicate const &predicate) {
+  std::vector<std::string> newVector;
+  newVector.reserve(toIterator - fromIterator);
+  std::transform(fromIterator, toIterator, std::back_inserter(newVector),
+                 predicate);
+  return newVector;
+}
+
+std::vector<std::string> getInputSuffixes(QStringList files) {
+  auto const allSuffixes =
+      transform(files.begin(), files.end(), [&](QString const &file) {
+        QFileInfo const fileInfo(file);
+        return extractLastOf(fileInfo.baseName().toStdString(), "_");
+      });
+
+  auto const suffixes = copyIf(allSuffixes.begin(), allSuffixes.end(),
```

Comment:
I'm not entirely sure that creating a wrapper around the std functions is helping to save code. I think we should just use the std functions wherever we need them